### PR TITLE
124 docs improvements

### PR DESCRIPTION
**Pull request by misterbisson**
This addresses documentation improvements requested in https://github.com/joyent/containerpilot/issues/124

### Big picture

- The canonical source for ContainerPilot documentation will move to joyent.com/containerpilot/docs (it's not there yet, this PR is part of that).
- Documentation will be maintained in this repo (the content of this PR).

### Content changes and structure

- The top-level readme will only appear in the repo. I’ve cleaned it up and written next text while mixing in some of the most salient bits of what you’d written back in the day
- The main docs move to /documentation.
- That dir contains Kirbified content to appear on Joyent.com at https://www.joyent.com/containerpilot/docs, with sub-pages based on the folder structure

### Kirby

The Kirby base/theme is in https://github.com/joyent/kirby-preview-base. That's a generic theme that intentionally has no Joyent branding, no Kirby license code...mostly nothing distinctive (though it is based on https://github.com/nsteiner/kdoc).

### Preview via Docker

```bash
docker build -t mydocs . 
docker run -d -p 80:80 --name mydocs mydocs
```

/cc @tgross @justenwalker

### CODE REVIEW

File: documentation/20-faq/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+### I've been told that multi-process containers are bad. Is that true?
+
+There are a number of conveniences in a container that does just one thing (offers a single service from a single app). It means that starting the container is the same as starting the app in it, and sending a signal to that container to shut down is the same as stopping the application. These containers offer a number of advantages for "modern" applications and operations, including automation of the application lifecycle.
+
+However, it's hardly the only "right" way to build a container. ContainerPilot is optimized for containers that offer a single service, but [it can be used in containers and VMs running any number of services](#where-does-containerpilot-run). ContainerPilot makes absolutely no judgements about how you architect your application.
+
+### Does ContainerPilot support IPv6?
+
+It's not
```
Comment (tgross) [inline documentation/20-faq/README.md]: This isn't finished.

File: documentation/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+```bash
+docker build -t mydocs
+docker run -d -p 80:80 --name mydocs mydocs
+```
+
+The Dockerfile uses https://github.com/joyent/kirby-preview-base ([in Docker Hub](https://hub.docker.com/r/joyent/kirby-preview-base/)) for the base Kirby install and theme.
+
+### Language styleguide
+
+- Please do not force wrap text in `*.md` files
```
Comment (tgross) [inline documentation/CONTRIBUTING.md]: ❤️

File: integration_tests/tests/test_telemetry/run.sh [hunk 3]
```diff
@@ -1,4 +1,13 @@
 #!/bin/bash
```
Comment (tgross) [inline integration_tests/tests/test_telemetry/run.sh]: Not sure why we'd see this commit in this branch. Where'd this fork from?

File: documentation/12-configuration/README.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+There are a few ways to specify the ContainerPilot configuration location.
+
+1. An environment variable set in the `docker run...` string, in a Docker Compose manifest, in the Dockerfile, or elsewhere
+2. As an argument to ContainerPilot, perhaps in the the `docker run...` string, or in in the `CMD` or `ENTRYPOINT` of the Dockerfile
+3. As a JSON string passed to ContainerPilot in any of the locations named above
+
+#### Examples: specifying the configuration file path
+
+```bash
```
Comment (tgross) [inline documentation/12-configuration/README.md]: The syntax highlighting for this is displaying black-on-black in the preview container.

File: documentation/13-service-discovery/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      "name": "{{ .BACKEND }}",
+      "poll": 7,
+      "onChange": "/usr/local/bin/reload.sh"
+    }
+  ],
+```
+
+That script executes the the following when called for that event:
+
+```bash
```
Comment (tgross) [inline documentation/13-service-discovery/README.md]: This is displaying black-on-black in the preview container.

**Comment by tgross** [comment]
It looks like everywhere we're using the syntax highlighting hint `bash`, we're getting black-text-on-dark-background when I preview it.

File: documentation/13-service-discovery/README.md (hunk 5 above)
Comment (misterbisson) [inline documentation/13-service-discovery/README.md]: Opened issue in https://github.com/joyent/kirby-preview-base/issues/3

File: documentation/12-configuration/README.md (hunk 4 above)
Comment (misterbisson) [inline documentation/12-configuration/README.md]: Opened issue in https://github.com/joyent/kirby-preview-base/issues/3

File: integration_tests/tests/test_telemetry/run.sh (hunk 3 above)
Comment (misterbisson) [inline integration_tests/tests/test_telemetry/run.sh]: This was a surprise that emerged from my commit squashing. I'm not clear what git sorcery caused it.

File: documentation/13-service-discovery/README.md (hunk 5 above)
Comment (misterbisson) [inline documentation/13-service-discovery/README.md]: We ship different CSS for the joyent.com content. Once that's settled, I'll circle back on https://github.com/joyent/kirby-preview-base/issues/3 to update and correct it.

**Pull request merged**